src/details/ArborX_DetailsLegacy.hpp

```diff
@@ -35,8 +38,12 @@ class LegacyValues
       : _primitives(primitives)
   {}
 
+  LegacyValues(Primitives const &primitives, BoundingVolume const &)
+      : _primitives(primitives)
+  {}
```

Comment:
Yes, this is to avoid specializing template arguments. There are a lot of situations when we can't deduce it, like constructing a hierarchy on kDOPs, where primitives would just return a point or a box.

Reply:
> I'm confused. Removing this constructor causes no compiler errors for me. Where is it used?

Not used at the moment. It's only there for the detection guide, that's not currently used.

Reply:
> Given the number of comments about the thing, I should just remove it.

Yes, I would prefer that.

Reply:
I had no particular issue about that constructor.  I was going to suggest deduction guidelines for the most common case w/o necessarily removing that one.